agent/engine/task_manager_test.go

```diff
@@ -249,12 +248,12 @@ func TestStartContainerTransitionsInvokesHandleContainerChange(t *testing.T) {
 	defer containerChangeEventStream.Unsubscribe(eventStreamName)
 
 	go func() {
-		<-containerEvents
+		<-stateChangeEvents
```

Comment:
Can you combine the two go routines into a single one? You'll also have to modify `eventsGenerated.Add(3)` to `eventsGenerated.Add(2)` as well.

```
go func() {
 <-stateChangeEvents
 <-stateChangeEvents
 eventsGenerated.Done()
}()